spring-batch-infrastructure/src/main/java/org/springframework/batch/support/transaction/TransactionAwareBufferedWriter.java

```diff
@@ -72,6 +99,47 @@ public void setEncoding(String encoding) {
 		this.encoding = encoding;
 	}
 
+	/**
+	 * Buffers the output on a {@code byte[]} level. Compared to
+	 * buffering on a {@code char[]} level this saves about 50% memory
+	 * for common encodings and input on Java 8.
+	 */
+	static final class Buffer {
+
+		private ByteArrayOutputStream outputStream;
+
+		private OutputStreamWriter writer;
+
+		Buffer(String encoding) throws UnsupportedEncodingException {
+			this.outputStream = new ByteArrayOutputStream();
+			this.writer = new OutputStreamWriter(outputStream, encoding);
+		}
+
+		int length() throws IOException {
+			this.writer.flush();
+			return this.outputStream.size();
+		}
+
+		void append(char[] cbuf, int off, int len) throws IOException {
+			this.writer.write(cbuf, off, len);
+		}
+
+		void append(String str, int off, int len) throws IOException {
+			this.writer.write(str, off, len);
+		}
+
+		byte[] getBytes() throws IOException {
```

Comment:
This method is not used and can be removed.

Reply:
fixed